staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -426,10 +426,41 @@ func (g *Cloud) AddAliasToInstance(nodeName types.NodeName, alias *net.IPNet) er
 // Gets the named instances, returning cloudprovider.InstanceNotFound if any
 // instance is not found
 func (g *Cloud) getInstancesByNames(names []string) ([]*gceInstance, error) {
+	instanceOrErrors, err := g.getInstanceOrErrorsByNames(names)
+	if err != nil {
+		return nil, err
+	}
+	var failed []string
+	var allInstances []*gceInstance
+	for name, entry := range instanceOrErrors {
+		if entry.err == cloudprovider.InstanceNotFound {
+			failed = append(failed, name)
+		} else if entry.err != nil {
+			return nil, entry.err
+		} else {
+			allInstances = append(allInstances, entry.instance)
+		}
+	}
+	if len(failed) > 0 {
```

Comment:
looks like other `entry.err` types are not even possible when looking at the other function so I'm not sure what the reasoning for this is

Reply:
For extensibility InstanceNotFound may not be the only error to return in the future. The idea here is that an error for one instance does not prevent any other instances from being listed. I don't think leaving room for that makes the code much more complex.

Another reasoning is that instead of just found / not found, the error enum should also be returned for clarity.